apps/server/src/modules/tool/service/external-tool.service.spec.ts

```diff
@@ -246,6 +251,21 @@ describe('ExternalToolService', () => {
 
 			expect(result).toEqual({ data: [{ ...externalToolDO, config: oauth2ToolConfigDO }], total: 1 });
 		});
+
+		it('should filter out oauth2tools with unresolved oauthConfig when oauthProvider throws an error', async () => {
```

Comment:
test structure in generall with describe('when ...) const setup() { // include all what is needed for the test setup }

Reply:
I changed the structure, some tests share the setup in the describe and I think it makes sense this way  :)